Automattic/jetpack
# Mailchimp block: do not fetch groups when a site is not connected

## Proposed changes:

When a site is not connected to Mailchimp yet, we should not even attempt to fetch groups.

Related conversation: p1723665699956539-slack-C4N88L95W
Related diff: D158488-code

### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Create a Mailchimp account
* Create a new audience (mailing list)
* Go to Audience > Manage Audience > Groups
* Create a new group
* Go to https://wordpress.com/marketing/connections/yoursite.com and connect your site to the Mailchimp mailing list you just created
* Go to Posts > Add New
* Add a Mailchimp block
* You should see the group you created in the block sidebar.
